Added: shortcut for delete action in editor

dicene opened the pull request:
Exactly what the title says. Uses OS's default DELETE shortcut.

dicene commented:
Gonna see if there's an easy way to turn the QKeySequence into text so there is no potential situation where I'm indicating the wrong key.

dicene commented:
http://doc.qt.io/qt-4.8/qkeysequence.html#toString looks like the answer

dicene commented:
```c++
deleteTriggerAction->setToolTip(QStringLiteral("<html><head/><body><p>%1 (%2)</p></body></html>").arg(tr("Delete Item"), QKeySequence(QKeySequence::Delete).toString()));
```
gives me

http://puu.sh/wfMad/554ca598df.png

That look good to you, @vadi2?

dicene commented:
Uh, this is somehow eating DEL presses from the edbee editor as well, so that's definitely a problem. :no_good_man:

vadi2 commented:
Maybe in the event for it, check that the selection is actually on the list of things and only act on it then, otherwise pass the event through?

dicene commented:
Fixed the issue with its shortcutContext being application-wide. Now it only fires when a treewidget is focused. Also updated the tool-tip ala the screenshot above. Should be good to go.

vadi2 commented:
It's pretty trivial so I'll go ahead with the merge, but if you see any issues @SlySven let us know!
